### PR TITLE
Use new home API

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: doc/user/options.md [hunk 1]
```diff
@@ -125,7 +125,17 @@ Native switches:
...
 TruffleRuby needs to know where to locate files such as the standard library.
-These are stored in the TruffleRuby home directory. The TruffleRuby option
-`home` has priority for setting the home directory. Otherwise it is set
-automatically to the directory containing the TruffleRuby JAR file, if
-TruffleRuby is running on a JVM.
+These are stored in the TruffleRuby home directory.
+
+The search priority for finding Ruby home is:
+
+* The value of the TruffleRuby `home` option.
```
Comment (nirvdrum) [inline doc/user/options.md]: I think clarifying what you mean by "option" here would be helpful:
```
The value of the TruffleRuby `home` option (i.e., `-Xhome=path/to/home`).
```

File: src/main/java/org/truffleruby/RubyContext.java [hunk 2]
```diff
@@ -710,9 +710,24 @@ private String searchRubyHome(Options options) throws IOException {
             return home.getCanonicalPath();
         }
 
-        StringBuilder warning = new StringBuilder("home locations tried:\n");
+        // Use the Truffle reported home
+
+        final String truffleReported = language.getTruffleLanguageHome();
+
+        if (truffleReported != null) {
+            final File home = new File(options.HOME);
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/RubyContext.java]: Should this be `new File(truffleReported)`?

Comment (chrisseaton) [reply src/main/java/org/truffleruby/RubyContext.java]: Yes thanks that was the whole point of this PR!

File: doc/user/options.md (hunk 1 above)
Comment (chrisseaton) [reply doc/user/options.md]: Fixing.

File: doc/user/options.md [hunk 3]
```diff
@@ -125,7 +125,18 @@ Native switches:
...
+* The home that the Truffle framework reports.
+* The parent of the directory containing the Ruby launcher executable.
+* The value of the system property `graalvm.home` with `jre/languages/ruby`.
```
[the remark this answers is not in the document]
Comment (chrisseaton) [reply doc/user/options.md]: Fixing.

**Pull request merged**